packages/taco/src/sign.ts

```diff
@@ -0,0 +1,47 @@
+import { Domain, getPorterUris, PorterClient } from '@nucypher/shared';
+import { SigningCoordinatorAgent } from '@nucypher/shared';
+import { ethers } from 'ethers';
+
+import { ConditionContext } from './conditions/context';
+import { SigningOptions, SignResult, UserOperation } from './types';
+
+export async function signUserOp(
+  provider: ethers.providers.Provider,
+  userOp: UserOperation,
+  chainId: number,
+  aaVersion: "mdt" | "0.8.0" | string,
+  cohortId: number,
+  domain: Domain,
+  options: SigningOptions = { optimistic: true, returnAggregated: true },
+  context?: ConditionContext,
+  porterUris?: string[]
```

Comment:
Bit of a nit-pick, can we reorder the params? My thinking here is to go from more general -> more specific values

```suggestion
  provider: ethers.providers.Provider,
  domain: Domain,
  cohortId: number,
  chainId: number,
  userOp: UserOperation,
  aaVersion: "mdt" | "0.8.0" | string,]
  options: SigningOptions = { optimistic: true, returnAggregated: true },
  context?: ConditionContext,
  porterUris?: string[]
```